explore/marginal-gain/marginal-gain-multiple-genes-pca.py

```diff
@@ -0,0 +1,340 @@
+
+# coding: utf-8
+
+# # How much predictiveness does gene expression add over covariates for multiple mutations (using pca)? 
+
+# Some notes on the results displayed here:
+# 1) Uses multiple genes
+# 2) Analyzes training/testing auroc for 3 data sets
+#     -covariates only
+#     -expression only
+#     -covariates+expression
+# 3) Pca performed instead of feature selection.
+
+# In[41]:
+
+import seaborn as sns
+import matplotlib.pyplot as plt
+import os
+import pandas as pd
+import numpy as np
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler, Imputer
+from sklearn.linear_model import SGDClassifier
+from sklearn.model_selection import train_test_split, GridSearchCV
+from sklearn.metrics import roc_auc_score
+from sklearn.decomposition import PCA
+from matplotlib import gridspec
+from sklearn.linear_model import LinearRegression
+
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+
+from IPython.core.debugger import Tracer
+from IPython.display import display
+import warnings
+warnings.filterwarnings("ignore") # ignore deprecation warning for grid_scores_
+
+
+# In[42]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# In[43]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('..', '..', 'download', 'covariates.tsv')\ncovariates = pd.read_table(path, index_col=0)")
```

Comment:
Let's restrict to only the following covariates:

+ `n_mutations_log1p`
+ anything starting with `acronym_`

We've seen these are the most important. This way we can remove imputation, which won't currently work for categorical variables that have been dummied anyway.